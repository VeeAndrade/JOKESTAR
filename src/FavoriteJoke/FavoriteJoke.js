import React from 'react';
import './FavoriteJoke.scss';

const FavoriteJoke = (jokeObj) => {
  const { id, setup, joke, punchline, delivery } = jokeObj.jokeObj
  return (
    <article key={id}className='favorite-joke-card'>
      <h3 className='setup'>{setup || joke}</h3>
      <p className='punchline'>{punchline || delivery}</p>
    </article>
  )
}

export default FavoriteJoke;