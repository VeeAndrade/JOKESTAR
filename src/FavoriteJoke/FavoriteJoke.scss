.favorite-joke-card {
  color: #fff;
  background: rgba(58, 126, 27, 0.473);
  width: 550px;
  height: 150px;
}

.setup {
  color: black;
}