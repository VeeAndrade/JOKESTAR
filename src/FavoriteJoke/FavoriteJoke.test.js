import React from 'react';
import { render } from '@testing-library/react';
import FavoriteJoke from './FavoriteJoke';
import { shallow } from 'enzyme';

describe('FavoriteJoke', () => {
  it('should match the snapshot', () => {
    const mockJoke = {
      id:42,
      setup: 'how much does a polarbear weigh?',
      punchline: 'enough to break the ice'
    }
    const wrapper = shallow(<FavoriteJoke jokeObj={mockJoke}/>)
    expect(wrapper).toMatchSnapshot();
  })
})