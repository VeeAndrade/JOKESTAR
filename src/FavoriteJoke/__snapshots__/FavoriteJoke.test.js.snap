// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteJoke should match the snapshot 1`] = `
<article
  className="favorite-joke-card"
  key="42"
>
  <h3
    className="setup"
  >
    how much does a polarbear weigh?
  </h3>
  <p
    className="punchline"
  >
    enough to break the ice
  </p>
</article>
`;
